pyproject.toml

```diff
@@ -198,3 +198,45 @@ typeCheckingMode = "standard"
 # that is not meant to be exported from that module. The default value for this
 # setting is "error".
 reportPrivateImportUsage = false
+
+[tool.ruff]
+# exclude = ["__init__.py"]
+line-length = 120
+target-version = "py311"
+
+[tool.ruff.lint]
+select = ["I", "E", "W", "D", "PL"]
+# ruff configuration
+# E, W -- pycodestyle
+# D -- pydocstyle
+# I -- isort
+# PL -- pylint
+
+# Rules considered over other incompatible ones
+exclude = ["docs/conf.py", "setup.py"]
+
+ignore = [
+  "D407", "D203", "D213", "D416", "PLR0912", "PLR0911", "PLR0915", "PLR2004", "PLR0913",
+  "D103", "D100", "D101", "D102", "D105", "D104", "D107", "D205", "D301","D400", "D401", "D404", "D415", "D417", "E501",
+  "PLR1714", "PLC0414", "PLC0105", "PLW2901",
```

Comment:
Note that I plan to create subissues in #589 to disable these ignored rules one by one.